geom/wkt_lexer.go

```diff
@@ -1,54 +1,50 @@
 package geom
 
 import (
-	"errors"
-	"io"
+	"strings"
 	"text/scanner"
 )
 
 type wktLexer struct {
-	scn       scanner.Scanner
-	peeked    bool
-	nextToken string
+	scn    scanner.Scanner
+	peeked string
 }
 
-func newWKTLexer(r io.Reader) *wktLexer {
+func newWKTLexer(wkt string) wktLexer {
 	var scn scanner.Scanner
-	scn.Init(r)
-	return &wktLexer{scn: scn}
+	scn.Init(strings.NewReader(wkt))
+	return wktLexer{scn: scn}
 }
 
-func (w *wktLexer) next() (string, error) {
-	if w.peeked {
-		tok := w.nextToken
-		w.peeked = false
-		w.nextToken = ""
-		return tok, nil
+func (w *wktLexer) next() string {
+	if w.peeked != "" {
+		tok := w.peeked
+		w.peeked = ""
+		return tok
 	}
 
-	var err error
 	w.scn.Error = func(_ *scanner.Scanner, msg string) {
-		err = errors.New(msg)
+		panic(msg)
```

Comment:
TODO: need to investigate more in what cases this could happen (and make sure that it can't).

Reply:
Not using panic here anymore.